htdocs/compta/paiement/class/paiement.class.php

```diff
@@ -354,7 +354,7 @@ public function create($user, $closepaidinvoices = 0, $thirdparty = null)
 										// Loop on each vat rate
 										$i = 0;
 										foreach ($invoice->lines as $line) {
-											if ($line->total_ht != 0) {    // no need to create discount if amount is null
+											if ($line->product_type != 9 && $line->total_ht != 0) {    // no need to create discount if special product or amount is null
```

Comment:
Do you mean you have a line with
product_type = 9 and also total_ht that is not 0.

We can use product_type to identify special line, but we should keep the rule "total of lines of invoices in database must match total of invoice in database".  So if a special line exists and must not be included into total, a better practice would be to have total to 0 for such line;

Can you describe the use case / module that add such a line ?

Reply:
Yes, I have a line with "product_type" = 9 and "total_ht" is not 0.
For example with the "Ouvrage" module, you can create a deposit invoice with these lines : 
![image](https://user-images.githubusercontent.com/45359511/171570637-ce40b465-8115-4941-908c-affe1a58bc9e.png)

So when you put a payment, the amount for third-party discount is double counted
- one time in the first line (work line)
- and an other time in others lines (components lines of the work line)